rlberry/manager/agent_manager.py

```diff
@@ -545,6 +546,25 @@ def fit(self, budget=None, **kwargs):
         del kwargs
         budget = budget or self.fit_budget
 
+        # If spawn, test that protected by if __name__ == "__main__"
+        if self.mp_context == "spawn":
+            try:
+                _check_not_importing_main()
+            except RuntimeError as exc:
+                raise RuntimeError(
+                    """Warning: in AgentManager, if mp_context='spawn' and
+                        parallelization="process" then the script must be run *
+                        outside a notebook and protected by a  if __name__ == '__main__':
+                                   For instance :
+
+                                       agent = AgentManager(UCBVIAgent,(Chain, {}),
+                                                            mp_context="spawn",
+                                                            parallelization="process")
+                                       if __name__ == '__main__':
```

Comment:
In the example, I think it's better to put everything inside the `if __name__ == '__main__'`:

```python
if __name__ == '__main__':
    agent = AgentManager(
                       UCBVIAgent,
                      (Chain, {}),
                      mp_context="spawn",
                      parallelization="process")
    agent.fit(10)
```
otherwise the `AgentManager` constructor will be re-run in each new interpreter created by spawn.

Reply:
Good catch, I actually always used it with only the fit I did not know that there was much of a difference. Thanks.